src/NuGet.Core/NuGet.Protocol/Utility/FindPackagesByIdNupkgDownloader.cs

```diff
@@ -121,8 +121,17 @@ internal FindPackagesByIdNupkgDownloader(HttpSource httpSource, IEnvironmentVari
                 url,
                 async stream =>
                 {
-                    await stream.CopyToAsync(destination, token);
-                    ProtocolDiagnostics.RaiseEvent(new ProtocolDiagnosticNupkgCopiedEvent(_httpSource.PackageSource, destination.Length));
+                    try
+                    {
+                        await stream.CopyToAsync(destination, token);
+                        ProtocolDiagnostics.RaiseEvent(new ProtocolDiagnosticNupkgCopiedEvent(_httpSource.PackageSource, destination.Length));
+                    }
+                    catch
+                    {
+                        destination.Position = 0;
```

Comment:
Team Review: Check if position can be set

Reply:
If I use `if (destination.CanSeek) destination.Position = 0`, then when it's a non-seekable stream, the stream will be corrupted, and any exception or log message will not contain that error message. With this code the way it is, it means that customers might be "stream is not seekable" instead of "timeout" in their output. But they can complain to the application author (NuGet itself should never use a non-seekable stream), who can then fix their app.

Reply:
Additionally, there is an API design decision to be made:  Should we check `destination.CanSeek` at the top of the method and throw an `ArgumentException` when it's not seekable?

In theory, any customer who is copying from an extremely reliable network connection, to a non-seekable stream, is unlikely to have ever experienced the corrupted download. But if we start throwing because the stream is not seekable, then it will completely break their app.

So, while it feels extremely unlikely that this would break anyone, and it would be better to throw early, I'm being cautious in not doing so.  Please let me know if you disagree. If there's enough I'm happy to add the check at the top of the method. It's just that after 2 API breaking changes this month, I don't want yet another conversation with partners about API changes.